langlib/lang.function/src/main/java/org/ballerinalang/langlib/function/Call.java

```diff
@@ -0,0 +1,132 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.langlib.function;
+
+import io.ballerina.runtime.api.async.StrandMetadata;
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.types.Parameter;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.values.BFunctionPointer;
+import io.ballerina.runtime.internal.TypeChecker;
+import io.ballerina.runtime.internal.scheduling.Scheduler;
+import io.ballerina.runtime.internal.types.BArrayType;
+import io.ballerina.runtime.internal.types.BFunctionType;
+import io.ballerina.runtime.internal.types.BTupleType;
+import io.ballerina.runtime.internal.util.exceptions.BLangExceptionHelper;
+import io.ballerina.runtime.internal.util.exceptions.RuntimeErrors;
+import io.ballerina.runtime.internal.values.ArrayValueImpl;
+import io.ballerina.runtime.internal.values.ListInitialValueEntry;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_BUILTIN_PKG_PREFIX;
+import static io.ballerina.runtime.api.constants.RuntimeConstants.FUNCTION_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.INCOMPATIBLE_TYPES;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Native implementation of lang.function:call(function func, any|error... args).
+ *
+ * @since 2201.2.0
+ */
+public class Call {
+
+    private static final StrandMetadata METADATA = new StrandMetadata(BALLERINA_BUILTIN_PKG_PREFIX, FUNCTION_LANG_LIB,
+                                                                      "1.0.0", "call");
+
+    public static Object call(BFunctionPointer<Object, Object> func, Object... args) {
+        BFunctionType functionType = (BFunctionType) func.getType();
+        List<Type> paramTypes = new LinkedList<>();
+        List<Type> argTypes = new LinkedList<>();
+        List<Object> argsList = new java.util.ArrayList<>(List.of(Scheduler.getStrand()));
+
+        if (checkIsValidPositionalArgs(args, argsList, functionType, paramTypes, argTypes) ||
+                 checkIsValidRestArgs(args, argsList, functionType, paramTypes, argTypes)) {
+                throw ErrorCreator.createError(
+                        getModulePrefixedReason(FUNCTION_LANG_LIB, INCOMPATIBLE_TYPES),
+                        BLangExceptionHelper.getErrorDetails(RuntimeErrors.INCOMPATIBLE_TYPE,
+                                                             new BTupleType(paramTypes), new BTupleType(argTypes)));
```

Comment:
The error for 

```ballerina
public function main() {
    any|error x = function:call(fn, 1, 2, 3);
}

function fn(int a, string... b) {

}
```

is still

```
error: {ballerina/lang.function}IncompatibleTypes {"message":"incompatible types: expected '[int,string[]]', found '[int,int,int]'"}
```

Shall we properly model the rest param as the rest type?

```
error: {ballerina/lang.function}IncompatibleTypes {"message":"incompatible types: expected '[int,string...]', found '[int,int,int]'"}
```

Reply:
Fixed